### PR TITLE
Angular version 18 + removes unsupported packages

### LINKED ISSUE #176: Update to Angular 18
This will require an incremental increase in the Angular versions to get to 18. We will need to ensure that the dependencies are available and the functionality required remains.

### PR DESCRIPTION
**Pull request by BobTorgerson**
This PR increases the Angular version of the code base to Angular 18, while upgrading the packages that needed to be updated along the way. One package called ng2_smart_table no longer is supported and so this code also removes the dependency of that package and replaces it with the supported Angular Material suite of tools. This includes a "smart table" design that I've modified with CSS to look fairly close to the previous design. 

To test, you will want to increase your NPM to >= v18.19.0 and you may need to dump your node_modules directory.
`nvm install v18.19.0`
`rm -rf node_modules`
`npm install`
`npm run start`

The design of the site should remain largely the same, but the versions of the underlying dependencies will be much improved. 

Closes #176

### CODE REVIEW

File: src/app/app.module.ts [hunk 1]
```diff
@@ -32,8 +35,10 @@ import { SearchService } from "./search.service";
 import { UrlService } from "./url.service";
 import { CscComponent } from "./csc/csc.component";
 import { ProjectResourceComponent } from "./project-resource/project-resource.component";
-import { Ng2SmartTableModule } from "ng2-smart-table";
-import { TitleLinkComponent } from "./title-link/title-link.component";
+import { MatTableModule } from "@angular/material/table";
+import { MatSortModule } from "@angular/material/sort";
+import { MatCheckboxModule } from "@angular/material/checkbox";
+import { BrowserAnimationsModule } from "@angular/platform-browser/animations";
```
Comment (brucecrevensten) [inline src/app/app.module.ts:41]: Is this used anywhere?

File: src/app/app.module.ts [hunk 2]
```diff
@@ -32,8 +35,10 @@ import { SearchService } from "./search.service";
 import { UrlService } from "./url.service";
 import { CscComponent } from "./csc/csc.component";
 import { ProjectResourceComponent } from "./project-resource/project-resource.component";
-import { Ng2SmartTableModule } from "ng2-smart-table";
-import { TitleLinkComponent } from "./title-link/title-link.component";
+import { MatTableModule } from "@angular/material/table";
+import { MatSortModule } from "@angular/material/sort";
+import { MatCheckboxModule } from "@angular/material/checkbox";
```
Comment (brucecrevensten) [inline src/app/app.module.ts:40]: Is this used anywhere?

File: src/app/app.module.ts [hunk 3]
```diff
@@ -50,20 +55,22 @@ import { BreadcrumbComponent } from "./breadcrumb/breadcrumb.component";
...
     routing,
     NgSelectModule,
-    Ng2SmartTableModule,
     NgbDropdownModule,
     NgbModule,
     FontAwesomeModule,
+    MatTableModule,
+    MatSortModule,
+    MatCheckboxModule,
+    BrowserAnimationsModule,
```
Comment (brucecrevensten) [inline src/app/app.module.ts:73]: Remove if unused

File: src/app/app.module.ts [hunk 4]
```diff
@@ -50,20 +55,22 @@ import { BreadcrumbComponent } from "./breadcrumb/breadcrumb.component";
...
   ],
   providers: [
     GoogleAnalyticsService,
```
Comment (brucecrevensten) [inline src/app/app.module.ts:76]: This cannot be in use!  Either delete now or create ticket.

File: .browserslistrc [hunk 5]
```diff
@@ -15,3 +15,5 @@ not dead
 not IE 9-11 # For IE 9-11 support, remove 'not'.
 not op_mob <= 73
 not android <= 109
+not kaios 2.5
+not op_mini all
```
Comment (brucecrevensten) [inline .browserslistrc:19]: What are these doing?  When I start the app I get:

```
One or more browsers which are configured in the project's Browserslist configuration will be ignored as ES5 output is not supported by the Angular CLI.
Ignored browsers: kaios 2.5, op_mini all
```

File: src/app/csc/csc.component.html [hunk 6]
```diff
@@ -70,79 +65,62 @@ <h4 class="no_data">
...
-        <div class="d-none d-lg-block">
-          <ng2-smart-table
-            class="table table-striped"
-            [settings]="settings"
-            [source]="filteredCscProjectsList"
-          ></ng2-smart-table>
-        </div>
-      </div>
-    </div>
-    <!-- ********************< Mobile Cards*****************************************************-->
```
Comment (brucecrevensten) [inline src/app/csc/csc.component.html:82]: I don't think we should be removing this for mobile functionality -- do an A/B between this branch and production to see why (in mobile emulation mode)

File: src/app/csc/csc.component.html [hunk 7]
```diff
@@ -3,46 +3,41 @@
   <h2>{{ title }} Projects</h2>
   <div class="row csc">
     <div class="col-md-2">
+      <!-- Filter Section -->
       <div class="select-box">
         <h3 class="select-title">Topic</h3>
         <div *ngFor="let topic of topics">
-          <input
-            type="checkbox"
+          <mat-checkbox
```
Comment (brucecrevensten) [inline src/app/csc/csc.component.html:10]: These need to be formatted to look like the current production site

File: src/app/csc/csc.component.html [hunk 8]
```diff
@@ -70,79 +65,62 @@ <h4 class="no_data">
...
-                {{ childProject.title }}
-              </li>
-            </ul>
-          </h4>
-
-          <!--Project Year-->
-          <div class="card-text">
-            <b>Funding Year: </b>
-            <span *ngIf="project['fiscal_year']">
+        <table
```
Comment (brucecrevensten) [inline src/app/csc/csc.component.html:68]: Formatting needs to match production site

File: src/app/csc/csc.component.ts [hunk 9]
```diff
@@ -92,32 +78,24 @@ export class CscComponent implements OnInit {
...
-    for (var project in this.cscProjectsList) {
-      this.filteredCscProjectsList.push(this.cscProjectsList[project]);
-    }
+    private cdr: ChangeDetectorRef,
+  ) {
+    this.dataSource = new MatTableDataSource<any>();
   }
 
-  sortProjectsByKey(array, key) {
-    return array.sort(function (a, b) {
```
Comment (brucecrevensten) [inline src/app/csc/csc.component.ts:105]: yessssss bye

File: src/app/shared.scss [hunk 10]
```diff
@@ -3,32 +3,32 @@ h2 {
...
-    padding-bottom: 2px;
-    border-bottom: 1px solid #eee;
-  }
-  div {
-    margin: 0 0 0.2rem 0;
-    padding-left: 1.1rem;
-    text-indent: -1.1rem;
-    line-height: 1.4;
-    font-size: 1rem;
+  .mat-checkbox {
```
Comment (brucecrevensten) [inline src/app/shared.scss:6]: See prior comment/notes, these need to be adjusted to look like the production version

File: src/app/topics/topics.component.html [hunk 11]
```diff
@@ -86,89 +90,62 @@ <h4 class="no_data">
...
       <div *ngIf="!dataLoading && filteredProjectsList.length > 0">
-        <div class="d-none d-lg-block">
-          <ng2-smart-table
-            class="table table-striped"
-            [settings]="settings"
-            [source]="filteredProjectsList"
-          ></ng2-smart-table>
-        </div>
-
-        <!-- Mobile -->
```
Comment (brucecrevensten) [inline src/app/topics/topics.component.html:97]: Same as before, the diff is a little hairy here but looks like the code for mobile card-based view was dropped.  Restore!

**Review by brucecrevensten — CHANGES_REQUESTED** [review]
Great progress!!  A few detailed notes sprinkled throughout but the key pieces to complete are:

 - get the formatting of the new table components to match the existing ones as closely as possible,
 - same for checkboxes and headers
 - restore mobile view as cards

File: src/app/app.module.ts (hunk 3 above)
Comment (BobTorgerson) [reply src/app/app.module.ts:72]: Those are all used for Angular Material for both the tables and checkboxes.

File: src/app/app.module.ts (hunk 2 above)
Comment (BobTorgerson) [reply src/app/app.module.ts:39]: Yes, in the csc and topics components. The TitleLinkComponent was not used anywhere.

File: src/app/app.module.ts (hunk 1 above)
Comment (BobTorgerson) [reply src/app/app.module.ts:40]: Yes

File: src/app/app.module.ts (hunk 4 above)
Comment (BobTorgerson) [reply src/app/app.module.ts:75]: What do you mean they cannot be in use? They are all used to replace the ng2_smart_table and checkbox package that was deprecated.

File: .browserslistrc (hunk 5 above)
Comment (BobTorgerson) [reply .browserslistrc:19]: Those changes should get rid of those errors which is why they are included.

File: src/app/csc/csc.component.html (hunk 7 above)
Comment (BobTorgerson) [reply src/app/csc/csc.component.html:10]: Done!

File: src/app/csc/csc.component.html (hunk 8 above)
Comment (BobTorgerson) [reply src/app/csc/csc.component.html:68]: Done!

File: src/app/csc/csc.component.ts (hunk 9 above)
Comment (BobTorgerson) [reply src/app/csc/csc.component.ts:105]: I thought you might be happy about that 😄

File: src/app/topics/topics.component.html (hunk 11 above)
Comment (BobTorgerson) [reply src/app/topics/topics.component.html:97]: Added back in! Thanks for catching this

File: src/app/csc/csc.component.html (hunk 6 above)
Comment (BobTorgerson) [reply src/app/csc/csc.component.html:82]: Added back in! Thanks for catching this

File: src/app/app.module.ts (hunk 4 above)
Comment (brucecrevensten) [reply src/app/app.module.ts:75]: `GoogleAnalyticsService` can't be in use

File: src/app/search-nav/search-nav.component.ts [hunk 12]
```diff
@@ -145,7 +143,9 @@ export class SearchNavComponent implements OnInit {
     if (this.filteredStatus) {
       payload["status"] = this.combineLabels(this.filteredStatus);
     }
-    this.umamiService.eventEmitter("Search Submission", payload);
+    if (typeof umami !== "undefined") {
```
Comment (brucecrevensten) [inline src/app/search-nav/search-nav.component.ts:146]: Maybe we can use the lovely: `umami?.track()` syntax here?

**Review by brucecrevensten — APPROVED** [review]

**Pull request merged**